src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -0,0 +1,1072 @@
+        mStorage->Release();
+        mStorage = nullptr;
+    }
+
+    CHIP_ERROR GetAuthMode(AuthMode & authMode) const override
+    {
+        authMode = mStorage->mAuthMode;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetFabricIndex(FabricIndex & fabricIndex) const override
+    {
+        fabricIndex = mStorage->mFabricIndex;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetPrivilege(Privilege & privilege) const override
+    {
+        privilege = mStorage->mPrivilege;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR SetAuthMode(AuthMode authMode) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        mStorage->mAuthMode = authMode;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR SetFabricIndex(FabricIndex fabricIndex) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        mStorage->mFabricIndex = fabricIndex;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR SetPrivilege(Privilege privilege) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        mStorage->mPrivilege = privilege;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetSubjectCount(size_t & count) const override
+    {
+        count = 0;
+        for (const auto & subject : mStorage->mSubjects)
+        {
+            if (subject.IsEmpty())
+            {
+                break;
+            }
+            count++;
+        }
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetSubject(size_t index, NodeId & subject) const override
+    {
+        if (index < EntryStorage::kMaxSubjects)
+        {
+            return mStorage->mSubjects[index].Get(subject);
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+    CHIP_ERROR SetSubject(size_t index, NodeId subject) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        if (index < EntryStorage::kMaxSubjects)
+        {
+            return mStorage->mSubjects[index].Set(subject);
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+    CHIP_ERROR AddSubject(size_t * index, NodeId subject) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        size_t count;
+        GetSubjectCount(count);
+        if (count < EntryStorage::kMaxSubjects)
+        {
+            CHIP_ERROR err = mStorage->mSubjects[count].Add(subject);
+            if (err == CHIP_NO_ERROR && index != nullptr)
+            {
+                *index = count;
+            }
+            return err;
+        }
+        return CHIP_ERROR_BUFFER_TOO_SMALL;
+    }
+
+    CHIP_ERROR RemoveSubject(size_t index) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        size_t count;
+        GetSubjectCount(count);
+        if (index < count)
+        {
+            const auto & dest = mStorage->mSubjects + index;
+            const auto & src  = dest + 1;
```

Comment:
These would be a lot clearer as `const auto *` instead of reference-to-pointer, which hides the fact that they are pointers and why the `+1` makes any sense.

Reply:
OK, done.